### ISSUE TITLE
some questions about finetuning this model on custom dataset

### ISSUE THREAD

**Issue by booker-max** [yuliangguo/depth_any_camera#11]
To improve the model's generalization on my dataset, I chose to fine-tune the model based on the pretrained DAC ResNet101 outdoor model located at /mnt/csi-data-aly/shared/public/bookerjin/ckpts/DAC/dac_resnet101_outdoor.pt. Throughout my experiment, I came to the following two conclusions: the fine-tuning code is consistent with the original training code, with the only change being a smaller learning rate.

1. For fisheye cameras with minimal distortion, the fine-tuning process works fine. The loss converges properly, and the results are satisfactory.

![Image](https://github.com/user-attachments/assets/b94549e6-d7e6-4121-8a25-788b3a2dfbef)

2. However, for fisheye cameras with significant distortion, the fine-tuning process fails to converge, and the final results are poor. I find this somewhat perplexing.

![Image](https://github.com/user-attachments/assets/1362e0f3-d4d8-4c6a-800e-1869cf81b5a0)

Do you have any suggestions for improving this?
Thank you.

**Comment by yuliangguo** [comment]
Thanks for trying. There seems to be a few new problems involved.

1. For the fine-tuning process, can you provide the visualizations out of
the dataloader so that I can have a sense of the data you feed into the
network for training?

2. Comparing the two cases, I feel the major difference is the content gap
from the pre-training datasets. The closer the gap is, the better the
fine-tune could be. One thing you could try is using the camera's pitch
angle in both training and testing. It might lead to some FoV loss and
involves some heuristic in cutting the valid area, but the ground-facing
data can look more similar to the pre-training data. It may lead to some
improvements.

3. Since none of the pre-training data, or the fine-tuning data has
ground-truth on those ground areas, those areas might be hopeless at
current setup. You may try to search for other existing data including
those areas, and mix them for the training. Our framework should support
such mix training, but a lot of extra work is involved.

4. How much training data do you have, can training from scratch work? If
there is enough ground-truth for the training, no need to fine-tune if you
just care about the narrow area covered by GT already. But I feel the goal
might be more about getting those No-GT area work. Then the key question is
still how to tackle 3.

**Comment by wcyjerry** [comment]
@booker-max May I ask how you get the depth gt for your customer dataset. And @yuliangguo if I have a sparse depth generate from lidar points, would it be enough to do the finetuning?

**Comment by yuliangguo** [comment]
Sparse lidar is OK for the fine tune, because supervision is applied
pointwise.  But if they are too sparse per image you may need to change the
code to improve learning efficiency.
